backend/src/main/java/com/huellapositiva/domain/actions/UpdateProposalImageAction.java

```diff
@@ -0,0 +1,73 @@
+package com.huellapositiva.domain.actions;
+
+import com.huellapositiva.domain.model.entities.Proposal;
+import com.huellapositiva.domain.model.valueobjects.EmailAddress;
+import com.huellapositiva.domain.model.valueobjects.ProposalStatus;
+import com.huellapositiva.domain.repository.ProposalRepository;
+import com.huellapositiva.domain.service.EmailCommunicationService;
+import com.huellapositiva.domain.service.ImageService;
+import com.huellapositiva.domain.service.RemoteStorageService;
+import com.huellapositiva.infrastructure.orm.entities.JpaContactPerson;
+import com.huellapositiva.infrastructure.orm.entities.JpaProposal;
+import com.huellapositiva.infrastructure.orm.repository.JpaContactPersonRepository;
+import com.huellapositiva.infrastructure.orm.repository.JpaProposalRepository;
+import lombok.AllArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+import javax.persistence.EntityNotFoundException;
+import java.io.IOException;
+import java.net.URL;
+import java.nio.file.AccessDeniedException;
+
+@Service
+@AllArgsConstructor
+public class UpdateProposalImageAction {
+
+    @Autowired
+    private JpaContactPersonRepository jpaContactPersonRepository;
+
+    @Autowired
+    private JpaProposalRepository jpaProposalRepository;
+
+    @Autowired
+    private ProposalRepository proposalRepository;
+
+    @Autowired
+    private EmailCommunicationService emailCommunicationService;
+
+    @Autowired
+    private JpaContactPersonRepository contactPersonRepository;
+
+    @Autowired
+    private ImageService imageService;
+
+    private final RemoteStorageService remoteStorageService;
+
+    public void execute(MultipartFile photo, String accountId, String proposalId) throws IOException {
+        imageService.validateProfileImage(photo);
```

Comment:
Validate profile image?

Reply:
Ok, we created a new method to validate proposal images